python/fledge/services/core/api/plugins/config_validator.py

```diff
@@ -215,15 +217,109 @@ def is_url(self, value):
         )
         return bool(url_pattern.match(value))
     
-    async def ping_host(self, hostname):
+    def _detect_container_environment(self):
         """
-        Perform host reachability test using socket connection.
+        Detect if running in a containerized environment.
         
-        This method uses socket-based connectivity testing instead of ICMP ping
-        to work in containerized environments where ping may not be available.
+        Returns:
+            bool: True if running in a container, False otherwise
+        """
+        try:
+            # Check for Docker container indicators
+            with open('/proc/1/cgroup', 'r') as f:
+                content = f.read()
+                if 'docker' in content or 'container' in content:
+                    return True
+            
+            # Check for container environment variables
+            import os
+            container_vars = ['DOCKER_CONTAINER', 'KUBERNETES_SERVICE_HOST', 'container']
```

Comment:
will this work for Podman too? @gnandan you did some work around Podman.

Reply:
removed _detect_container_environment() method. So now it's generic code for all